File: cpu_usage_indicator/python/cpu_usage_indicator.py
#!/usr/bin/env python3

import time

from led import Led
from proc_stat import ProcStat

class App:
	def __init__(self):
		self.stat = ProcStat()
		self.ledr = Led("/sys/class/leds/sunxi_led0r/brightness" , 0)
		self.ledg = Led("/sys/class/leds/sunxi_led0g/brightness" , 0)

	def run(self):
		time.sleep(0.8)

		self.ledr.set_brightness(0)
		self.ledg.set_brightness(0)

		time.sleep(0.2)

		self.usage = self.stat.get_cpu_usage()
		print("\rcpu usage = %d %%  " % self.usage, end = "")

		
		if self.usage < 30:
			self.ledg.set_brightness(30)
		elif self.usage < 60:
			self.ledg.set_brightness(5)
		elif self.usage < 90:
			self.ledr.set_brightness(15)
		else:
			self.ledr.set_brightness(150)


app = App()

try:
	while 1:
		app.run()
except KeyboardInterrupt:
	print("")
	del app

File: cpu_usage_indicator/java/CpuUsageIndicator.java
import java.io.IOException;
import java.lang.Thread;
import sun.misc.SignalHandler;
import sun.misc.Signal;


public class CpuUsageIndicator {
	public static void main(String[] args) throws IOException, InterruptedException {
		App app = new App();

		app.run();
	}
}

class App {
	public App() throws IOException{
		stat = new ProcStat();
		ledr = new Led("/sys/class/leds/sunxi_led0r/brightness");
		ledg = new Led("/sys/class/leds/sunxi_led0g/brightness");
		running = true;

		SignalHandler handler = new SignalHandler() { 
			public void handle(Signal sig) { 
				running = false;
			} 
		}; 
		Signal.handle(new Signal("INT"),handler); 
	}

	public void run() throws IOException, InterruptedException{
		while(running) {
			Thread.sleep(800);

			ledr.setBrightness(0);
			ledg.setBrightness(0);

			Thread.sleep(200);

			usage = stat.getCpuUsage();
			System.out.printf("\rcpu usage = %d %%  ", usage);
			System.out.flush();

			if(usage < 30) {
				ledg.setBrightness(30);
			} else if(usage < 60) {
				ledg.setBrightness(5);
			} else if(usage < 90) {
				ledr.setBrightness(15);
			} else {
				ledr.setBrightness(150);
			}
		}

		System.out.printf("\n");
		ledr.setBrightness(0);
		ledg.setBrightness(0);
	}

	private class ExitHandler extends Thread {
		public ExitHandler() {
			super("Exit Handler");
		}

		public void run() {
			running = false;
		}
	}

	public boolean running;

	int usage;
	ProcStat stat;
	Led ledr;
	Led ledg;
}
